### PR TITLE
[WIP] Test Coverage for Line Methods, Changes to Line Methods

### PR DESCRIPTION
**Pull request by marty-larocque**
Primarily, this PR addresses issue #186 (test coverage). It adds tests for `base.assert_line()` and `base.assert_lines_of_type()`. These tests should also provide coverage for `base.get_slope_yintercept()`.

While working on these tests, a number of other issues were encountered. This PR should also address those:
- #235 and #236 (Issues with using seaborn)
- #237 (A small improvement)
- #238 (A case that produces an error)
- #262 (An edge case that produces an error)

### CODE REVIEW

**Comment by marty-larocque** [comment]
Currently, this PR is not ready to be merged for a few reasons:
- The way that #235 is currently addressed in this PR is just a quick and dirty fix. We should decide on our approach to this issue before this PR is merged.
- The way that #238 is currently addressed just suppresses the error. It is not a reliable fix. `assert_lines_of_type()` likely needs to be completely refactored. We should also decide on our approach here.
- There are currently no docstrings for the tests.
- The tests require seaborn, which is not included in the requirements.

**Comment by marty-larocque** [comment]
This PR has been cleaned up a bit:
- #238 is addressed in a better way
- Docstrings have been added to the line tests
- Seaborn has been added to the dev requirements


However:
- #235 is still unaddressed. The test `test_line_type_reg_one2one` is failing for this reason. I believe that Nathan is working on a fix for this issue. 
- #262 is an issue I discovered recently, and is not yet addressed. The test `test_line_type_one2one` is failing for this reason. I believe that this is because the test here represents an edge case in which no data is plotted, but a line is plotted. We should either change the test to represent a more common case, or we should address this issue.


Finally:
- #236 is still addressed by this PR
- #237 is still addressed by this PR

We should wait until Nathan's PR addressing #235 is merged before we consider merging this PR. We should also address #262, one way or another, before we merge this PR.

File: matplotcheck/base.py
Comment (lwasser) [inline matplotcheck/base.py:1102]: lets:
1. Remove xtime as a parameter
2. Let's make the data extent (boolean) parameter to make this test OPTIONAL

**Review by lwasser — CHANGES_REQUESTED** [review]
Here - let's address the issue of making the xlims data test a parameter (boolean)
please also 
1. Update from master (so the pr passes)
2. Add a change log entry
3. Add any comments at the end for anything that hasn't been complete!

**Comment by marty-larocque** [comment]
@lwasser This PR should be ready to be merged. Just note that #262 still needs to be addressed. That is not anything urgent, and it's a fairly rare edge case, but it should be addressed at some point.

**Comment by lwasser** [comment]
TODO's here:
- [ ] is there another way in python to calculate a regression line? if so - should we add a test for that approach to ensure this works
- [ ] let's change regression option to linear-regression

think about edge cases...

**Comment by nkorinek** [comment]
@lwasser this can be closed due to this pr: https://github.com/earthlab/matplotcheck/pull/274

**Comment by lwasser** [comment]
ok thanks @nkorinek closing!!

**Pull request closed without merging**
